File: Binary_Tree/11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {

        int n = height.size();
        int start=0,end=n-1;
        int maxiwater=0;

        while(start<end){

            int mini = min(height[start],height[end]);
            maxiwater = max(maxiwater,mini*(end-start));

            if(height[start]<height[end]){
                start++;
            }
            else{
                end--;
            }
        }
        return maxiwater;
    }
};

File: Binary_Tree/11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {

        int water=0;
        int n = height.length;

        int start=0,end=n-1;

        while(start<end){

            int minilength = Math.min(height[start],height[end]);

            int width = end-start;

            water = Math.max(minilength*width,water);
            if(height[start]<height[end]){
                start++;
            }
            else{
                end--;
            }
        }
        return water;
        
    }
}